llvm/test/CodeGen/RISCV/cheri/ptradd-dagcombine-infinite.ll

```diff
@@ -0,0 +1,63 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; This IR previously triggered an infinite DAGCombiner loop inside DAGCombiner::visitPTRADD().
+; Check that this no longer happens and that we can fold the two GEPs into a single ISD::PTRADD.
+; RUN: %riscv64_cheri_purecap_llc -mattr=+m -verify-machineinstrs -O0 < %s | FileCheck %s
```

Comment:
-O0 is a strange choice, especially if you're thinking about making the more complex combines be disabled for -O0. It's also the reason cannot_fold_2 gives extremely stupid code for what is `neg a0, a0`.

Reply:
I chose -O0 here since the original test case failed at -O0, but I agree this does not make much sense. I was thinking of disabling the reassociation for -O0 in a follow-up commit once you're happy with this change, but I can also do that as part of this PR.

Reply:
It can be a follow-up, I just don't want to see test cases being written that end up not testing what we want at non-O0